docroot/modules/humsci/hs_migrate/hs_migrate.module

```diff
@@ -42,3 +43,89 @@ function hs_migrate_config_pages_presave(ConfigPagesInterface $config_page) {
     Cache::invalidateTags(['migration_plugins']);
   }
 }
+
+
+/**
+ * Impliments hook_preprocess_form().
+ */
+function hs_migrate_preprocess_form(&$variables) {
+  $form_id = $variables['element']['#form_id'];
+  if ($form_id == 'media_image_edit_form') {
+    // Checks if this media is part of a migration and if so, prevent delete.
+    $media_id = \Drupal::routeMatch()->getParameter('media');
+    if ($media_id instanceof \Drupal\media\Entity\Media) {
+      $media_imported = _hs_migrate_media_imported($media_id->id());
+      if ($media_imported) {
+        \Drupal::messenger()
+          ->addWarning(t('This media can not be deleted since it contains imported & synced data.'));
+
+        $variables['#attached']['library'][] = 'hs_migrate/readonly';
+        $variables['attributes']['class'][] = 'messages--readonly';
+      }
+    }
+
+  }
+}
+
+/**
+ * Checks to see if a given mediaID is associated with a migration.
+ *
+ * @var int $mediaID
```

Comment:
One last thing quick `mediaID` to `media_id`

```suggestion
 * Checks to see if a given media_id is associated with a migration.
 *
 * @var int $media_id
```

Reply:
Gah, missed that one. OK, fixed. :)

Reply:
Thanks Marc!